ethpm/backends/registry.py

```diff
@@ -78,11 +82,23 @@ def parse_registry_uri(uri: str) -> RegistryURI:
     """
     Validate and return (authority, pkg name, version) from a valid registry URI
     """
+    from web3.auto.infura import w3
     validate_registry_uri(uri)
     parsed_uri = parse.urlparse(uri)
-    address, chain_id = parsed_uri.netloc.split(":")
+    address_or_ens, chain_id = parsed_uri.netloc.split(":")
+    ns = ENS.fromWeb3(w3)
+    if ns.address(address_or_ens):
+        address = ns.address(address_or_ens)
+        ens = address_or_ens
+    elif is_address(address_or_ens):
```

Comment:
Seems like you might want this one to come first in the `if/elif` ordering since it's a simpler check and might be the more common case.